ethereum/rpc/backend/backend.go

```diff
@@ -148,9 +148,13 @@ func (e *EVMBackend) EthBlockFromTendermint(
 
 		gasUsed += gas
 
-		msg, isEthTx := tx.(*evmtypes.MsgEthereumTx)
+		if len(tx.GetMsgs()) != 1 {
```

Comment:
is that len(tx.GetMsgs()) > 1

Reply:
can you use a for loop here too? Otherwise this could break in the future

Reply:
done